File: Practice/25/C++/code.cpp
#include <vector>
#include <iostream>
#include <random>
using namespace std; 
void print(vector<int> mass){
    for (int i = 0; i < mass.size(); i++)
    {
        cout << mass[i] << " ";
    }
    cout << "\n";    
}
bool is_sorted(vector<int> mass, bool des = true){
    int size = mass.size();;
    if (des){
        for (int i = 0; i < size-1; i++)
        {
            if (mass[i] > mass[i+1]) return false;
        }
    }else{
        for (int i = 0; i < size-1; i++)
        {
            if (mass[i] < mass[i+1]) return false;
        }
    }
    return true;
}
vector<int> Bozosort(vector<int> mass, bool des = true){
    int size = mass.size();
    vector<int> result = mass;
    while(is_sorted(result, des) == false){
        for (int i=0;i<size;i++){
            int k = std::rand() % size;
            int j = std::rand() % size;
            swap(result[k],result[j]);
        }
    }
    return result;
}
vector<int> Bozosort(vector<vector<int>> mass, bool des = true){
    vector<int> result;
    for (vector<int> vec : mass){
        for(int elem : vec){
            result.push_back(elem);
        }
    }
    return Bozosort(result, des);
}
vector<int> Bozosort(int a1, int a2, int a3, bool des=true){
    vector<int> result = {a1,a2,a3};
    return Bozosort(result, des);
}
int main(){
    int n;
    cin >> n;
    vector<int> navvod;
    vector<int> stroka;
    vector<vector<int>> matrica;
    for (int i = 1; i < n+1; i++)
    {
        int num;
        cin >> num;
        stroka.push_back(num);
        navvod.push_back(num);
        if (i % int(sqrt(n))==0){
            matrica.push_back(stroka);
            stroka.clear();
        }
    }
    print(Bozosort(navvod));
    print(Bozosort(navvod, false));
    print(Bozosort(matrica));
    print(Bozosort(matrica, false));
    print(Bozosort(navvod[0],navvod[1],navvod[2]));
    print(Bozosort(navvod[0],navvod[1],navvod[2],false));
}

File: Practice/25/Python/code.py
import random as rand
import time
import math
def BozoSort(mass, des = True):
    result = []
    if type(mass[0]) == list:
        for elem in mass:
            result += elem
    else:
        result = mass.copy()
    n = len(result)
    while(is_sorted(result, des) == False):
        for i in range(n-1):
            k, i = rand.randint(0,n-1), rand.randint(0,n-1)
            result[i], result[k] = result[k], result[i]
    return result
def is_sorted(mass, des):
    result = mass
    n = len(result)
    if des:
        for i in range (n-1):
            if result[i] > result[i+1]:
                return False
        return True
    else: 
        for i in range (n-1):
            if result[i] < result[i+1]:
                return False
        return True
def BozoSortF3(a,b,c,des = True):
    return BozoSort([a,b,c], des)
n = int(input())
while True:
    if int(math.sqrt(n)) != float(math.sqrt(n)) : 
        n =int(input("Plese, try again. The sqrt of nuber should be integer\n"))
    else:
        break
na = list(map(int, input().split()))
matrica = []
stroka = []
i = 0
for elem in na:
    stroka.append(elem)
    i +=1
    if i% math.sqrt(n) == 0:
        matrica.append(stroka)
        stroka = []
        i = 0
del stroka, i
del(na[n::])
if len(na) < n: print("Error, try again")
print(' '.join(map(str, BozoSort(na))))
print(' '.join(map(str, BozoSort(na, False))))
print(' '.join(map(str, BozoSort(matrica))))
print(' '.join(map(str, BozoSort(matrica, False))))
print(' '.join(map(str, BozoSortF3(na[0],na[1],na[2]))))
print(' '.join(map(str, BozoSortF3(na[0],na[1],na[2],False))))